test_conformance/api/test_kernel_private_memory_size.cpp

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "harness/errorHelpers.h"
+#include "harness/typeWrappers.h"
+#include <array>
+#include "procs.h"
+
+constexpr char const* ZERO_MEMORY_KERNEL =
+    R"(__kernel void private_memory( __global ulong *buffer , local ulong x[1]){)"
+    R"(    buffer[0] = x[0]; })";
+
+constexpr char const* ONE_MEMORY_KERNEL =
+    R"(__kernel void private_memory( __global ulong *buffer ){ )"
+    R"(    volatile __private ulong x[1];)"
+    R"(    buffer[0] = x[0]; })";
+
+constexpr char const* TWO_MEMORY_KERNEL =
+    R"(__kernel void private_memory( __global ulong *buffer ){ )"
+    R"(    volatile __private ulong x[1];)"
+    R"(    volatile __private ulong y[1];)"
+    R"(    buffer[0] = x[0] = y[0]; })";
```

Comment:
One of the benefits of raw string literals is that they preserve newlines, so you can avoid having to concatenate multiple literals together to build a kernel, which would make this more readable IMO. Example here:
https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/basic/test_intmath.cpp#L38